### PR TITLE
Quotas

### PR DESCRIPTION
**Pull request by cnuland**
Added the ability to create tshirt sized quotas which are managed by a cluster level resource

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,6 +5,24 @@ _This repository is currently undergoing active development. Functionality may b
...
+
+When the `quotaSize` attribute is defined in the `ProjectInitializeQuota` Custom Resource (CR) the operator will search for a cluster level `ProjectInitializeQuota` CR that defines a praticular quota size. This can be used to define predetermined t-shirt sizes when creating new projects (small, medium, large, etc)
+
+```
+apiVersion: redhatcop.redhat.io/v1alpha1
+kind: ProjectInitializeQuota
+metadata:
+  name: small
+spec:
+  resourceQuotaSpec:
```
Comment (sabre1041) [inline README.md]: Remove `resourceQuotaSpec`

File: deploy/role.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
+  creationTimestamp: null
   name: project-initialize
 rules:
 - apiGroups:
```
Comment (sabre1041) [inline deploy/role.yaml:7]: Considerations for `resourcequotas` must be made

**Review by sabre1041 — CHANGES_REQUESTED** [review]
@cnuland Good start. A few things to consider

1. Whenever a `ProjectInitializes` resource is created or detected, it should attempt to apply the appropriate quota to the project (either an add or update)
2. What happens when a `ProjectInitializeQuota` is updated? Does the change propagate to all projects that have it applied?
3. What happens when a Quota in a project that was created from a `ProjectInitializeQuota` is updated/modified? Does the configuration reset back to the expected state?
4. Do you want to add an owner reference of the quota resource to relate back to the `ProjectInitializeQuota` resource?

Interested to hear your thoughts

File: deploy/role.yaml (hunk 2 above)
Comment (cnuland) [reply deploy/role.yaml:7]: Added

File: README.md (hunk 1 above)
Comment (cnuland) [reply README.md]: Removed

**Review by sabre1041 — APPROVED** [review]

**Pull request merged**